provider/datagen/src/source.rs

```diff
@@ -34,11 +46,19 @@ impl SourceData {
     /// [GitHub downloads](https://github.com/unicode-org/icu/releases)).
     pub fn with_uprops(self, uprops_root: PathBuf) -> Self {
         Self {
-            cldr_paths: self.cldr_paths,
             uprops_root: Some(uprops_root),
+            ..self
         }
     }
 
+    /// Sets the [`TrieType`] to be used when generating data.
+    ///
+    /// For Unicode Properties data, the TrieType is implicitly set when selecting the
```

Comment:
Nit: clarify what this is used for if not for properties data. Or use it for properties data as well?